packages/core/src/state-management/state-manager.ts

```diff
@@ -323,12 +325,12 @@ export class StateManager {
       return
     }
 
-    await this._saveAnchorRequestForState(state$)
-    const anchorStatus$ = this.anchorService.requestAnchor({
-      streamID: state$.id,
-      tip: state$.tip,
-      timestampISO: (new Date()).toISOString(),
-    })
+    const genesisCID = state$.value.log[0].cid
+    const genesisCommit = await this.dispatcher.retrieveCommit(genesisCID, state$.id)
+    await this._saveAnchorRequestForState(state$, genesisCommit)
+
+    const carFile = await this._buildAnchorRequestCARFile(state$.id, state$.tip)
+    const anchorStatus$ = this.anchorService.requestAnchor(carFile)
```

Comment:
Looks like you will be retrieving the genesis commit from IPFS twice. Better just get it from the car file the second time.
```suggestion
    const carFile = await this._buildAnchorRequestCARFile(state$.id, state$.tip)
    const genesisCID = state$.value.log[0].cid
    const genesisCommit = carFile.get(genesisCID)
    await this._saveAnchorRequestForState(state$, genesisCommit)

    const anchorStatus$ = this.anchorService.requestAnchor(carFile)
```